src/walberla_bridge/src/BoundaryHandling.hpp

```diff
@@ -231,16 +231,18 @@ class BoundaryHandling {
   }
 #endif
 
-  Vector3<FloatType> get_force(IBlock *block) const {
+  Vector3<double> get_total_force(IBlock *block) const {
     return m_boundary->getForce(block);
   }
 
-  BlockDataID get_force_vector_id() const { return m_boundary->forceVectorID; }
-  BlockDataID get_index_vector_id() const { return m_boundary->indexVectorID; }
-  auto const &get_force_vector(const IBlock *block) {
+  auto const &get_force_vector(IBlock *block) const {
+    using ForceVector = BoundaryClass::ForceVector;
+    auto const force_vector_id = m_boundary->getForceVectorID();
+    auto *forceVector = block->getData<ForceVector>(force_vector_id);
+    forceVector->syncCPU();
```

Comment:
The call to `syncCPU()` was missing. The total LB force was correct, but the force per shape was always zero with the GPU kernels.